### ISSUE TITLE
allocator undefined for Proc

### ISSUE THREAD

**Issue by pawel2105** [puma/puma#1094]
On my server a capistrano task runs:

`~/.rvm/bin/rvm default do bundle exec pumactl -S /home/deploy/eapasa_deploy/shared/tmp/pids/puma.state restart`

This results in `allocator undefined for Proc`

This problem has been marked as resolved in previous issues but this is clearly not the case. Using puma (2.14.0) and rails (4.2.7.1)

My state file looks like this:

```

---
pid: 23422
config: !ruby/object:Puma::Configuration
  cli_options: 
  conf: 
  options:
    :min_threads: 8
    :max_threads: 32
    :quiet: true
    :debug: false
    :binds:
    - unix:///home/deploy/proj/shared/tmp/sockets/puma.sock
    :workers: 3
    :daemon: true
    :mode: :http
    :before_fork:
    - !ruby/object:Proc {}
    :worker_timeout: 60
    :worker_shutdown_timeout: 30
    :preload_app: true
    :directory: "/home/deploy/proj/current"
    :worker_directory: "/home/deploy/proj/current"
    :rackup: "/home/deploy/proj/current/config.ru"
    :environment: staging
    :pidfile: "/home/deploy/proj/shared/tmp/pids/puma.pid"
    :state: "/home/deploy/proj/shared/tmp/pids/puma.state"
    :redirect_stdout: "/home/deploy/proj/shared/log/puma_access.log"
    :redirect_stderr: "/home/deploy/proj/shared/log/puma_error.log"
    :redirect_append: true
    :prune_bundler: true
    :config_file: "/home/deploy/proj/shared/puma.rb"
    :control_auth_token: 8ceaa478db7beaa1abcf212b65215942
    :tag: '2016091601'

```

**Comment by pawel2105** [comment]
Removing the before_fork part removes the error:

```
:before_fork:
    - !ruby/object:Proc {}
```

**Comment by sriedel** [comment]
@pawel2105: I submitted a PR that fixes this issue and it got merged into puma 2.16.0. Can you see if the issue continues after you update your puma to a more recent version?

**Comment by terranisu** [comment]
@sriedel thanks a lot. after update to 2.16.0 (from 2.15.3) this problem is gone for me.
